.docker/Dockerfile

```diff
@@ -0,0 +1,291 @@
+# syntax=docker/dockerfile:1.7
+ARG DEV_FROM_STAGE=tooler
+ARG EXPORT_FROM_STAGE=builder
+ARG PREP_FROM_STAGE=runner
+ARG SHIP_FROM_STAGE=runner
+ARG WS_CACHE_ID=nav2
+
+ARG FROM_IMAGE=base
+# Stage from full image tag name for dependabot detection
+FROM ros:rolling AS base
+
+################################################################################
+# MARK: baser - setup base image using snapshots
+################################################################################
+FROM $FROM_IMAGE AS baser
+ENV FROM_IMAGE=${FROM_IMAGE}
+
+# Configure ubuntu snapshot
+RUN UBUNTU_DEB_SNAPSHOT=$(date -r /var/lib/dpkg/info +%Y%m%dT%H%M%SZ) && \
+  sed -i "s|http://archive.ubuntu.com/ubuntu/|http://snapshot.ubuntu.com/ubuntu/${UBUNTU_DEB_SNAPSHOT}|g" /etc/apt/sources.list && \
+  sed -i "s|http://security.ubuntu.com/ubuntu/|http://snapshot.ubuntu.com/ubuntu/${UBUNTU_DEB_SNAPSHOT}|g" /etc/apt/sources.list
+
+# Edit apt config for caching
+RUN mv /etc/apt/apt.conf.d/docker-clean /etc/apt/ && \
```

Comment:
Can you clarify why this is needed in the comment? Intent might be lost over the years